android/app/src/main/java/com/msupplymobile/Ble/BleManager.java

```diff
@@ -0,0 +1,202 @@
+package com.msupplymobile;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.WritableArray;
+import com.facebook.react.bridge.WritableMap;
+import android.util.Log;
+import java.util.Map;
+
+/**
+ * Singleton react-native class managing native Bluetooth adapter.
+ * Acts as a listener to both devices and a Bluetooth scanner.
+ * 
+ * React methods:
+ * getDevices: 
+ * Scans for devices using the DeviceScanner. Uses a manufacturerID (required)
+ * and a MAC address (optional) to filter results. Each device found is a BleDevice,
+ * which can be represented as a WritableMap to send back to JS.
+ * 
+ * sendCommand:
+ * Scans for a particular device using it's MAC address. Once found, the command is sent
+ * and calls back with the result - true/false. Downloaded data, if any, is held in the
+ * BleDevice.
+ */
+public class BleManager extends ReactContextBaseJavaModule implements BleScanListener, BleDeviceListener{
+    private ReactContext reactContext;
+    private Promise promise;
+    private String command;
+    private String macAddress;
+    private Map<String, BleDevice> devices;
+
+    public BleManager(ReactApplicationContext reactContext) {
+        super(reactContext);
+        this.reactContext = reactContext;
+    }
+
+    @Override
+    public String getName() {
+        return "BleManager";
+    }
+
+    /** 
+     * Callback for handling commands sent to devices,
+     * whether it was succcesful, or not.
+     * Overrides BleDeviceListener
+     */
+    @Override
+    public void BleCallback(boolean result){
```

Comment:
little bit iffy about this one, the only 'better' way i can think of here is to just return the device and have it as a single object in 'data'. So different return to JS if we are scanning for devices ({devices: [devices]}), or getting result of command for one device ({device: {single device info}} <- actually i think that would make more sense. what cha reckon ?